test/unit/test_common.py

```diff
@@ -78,3 +81,44 @@ def test_verify_checksum(input_file_name: str, content: str, expected_error: Exc
             verify_checksum(file_path)
     finally:
         shutil.rmtree(full_dir_path)
+
+DEFAULT_IMAGES = {
+    "HIP_VISIBLE_DEVICES": "quay.io/ramalama/rocm",
+}
+
+@pytest.mark.parametrize(
+    "accel_env,arg_override,env_override,config_override,expected_result",
+    [
+        (None, f"{DEFAULT_IMAGE}:latest", None, None, f"{DEFAULT_IMAGE}:latest"),
+        (None, None, f"{DEFAULT_IMAGE}:latest", None, f"{DEFAULT_IMAGE}:latest"),
+        (None, None, None, f"{DEFAULT_IMAGE}:latest", f"{DEFAULT_IMAGE}:latest"),
+        ("HIP_VISIBLE_DEVICES", None, None, None, "quay.io/ramalama/rocm:latest"),
+        ("HIP_VISIBLE_DEVICES", f"{DEFAULT_IMAGE}:latest", None, None, f"{DEFAULT_IMAGE}:latest"),
+        ("HIP_VISIBLE_DEVICES", None, f"{DEFAULT_IMAGE}:latest", None, f"{DEFAULT_IMAGE}:latest"),
+    ],
+)
+def test_accel_image(accel_env: str, arg_override: str, env_override: str, config_override: str, expected_result: str):
+    env = {"RAMALAMA_CONFIG": "/dev/null"}
+    if accel_env:
+        env[accel_env] = "1"
+    if env_override:
+        env["RAMALAMA_IMAGE"] = env_override
+
+    with patch.dict('os.environ', env, clear=True):
+        cmdline = []
+        if arg_override:
+            cmdline.extend(["--image", arg_override])
+        cmdline.extend(["run", "granite"])
+        parser = create_argument_parser("test_accel_image")
+        configure_subcommands(parser)
+        args = parser.parse_args(cmdline)
+
+        config = {}
+
+        load_config_from_env(config, os.environ)
```

Comment:
I think we can directly pass in `env` to `load_config_from_env` and not need to mock patch `os.environ`. This should make the code a bit simpler.

Reply:
I'm afraid it doesn't work, because the default value for `args.image` is [calculated](https://github.com/containers/ramalama/blob/cf43e310f0745e92a477217fb6f0e8ac478d66af/ramalama/cli.py#L121) using `acccel_image`, with `CONFIG` table; both refer to envvars such as RAMALAMA_IMAGE or HIP_VISIBLE_DEVICES (which may be set by ramalama itself).